src/Workspaces/Core/Portable/Shared/TestHooks/AsynchronousOperationListener.cs

```diff
@@ -17,6 +17,8 @@ internal sealed partial class AsynchronousOperationListener : IAsynchronousOpera
 
         private readonly string _featureName;
         private readonly HashSet<TaskCompletionSource<bool>> _pendingTasks = new HashSet<TaskCompletionSource<bool>>();
+        private CancellationTokenSource _blockedOnCompletionTokenSource;
```

Comment:
See other comments first, but please comment this.

Reply:
This should be clearer now, since the use of the cancellation token source is contained in same file with the source itself.